### PR TITLE
RichAudience - Fix bug multiple IMP & Add error with when video params is empty

### PR DESCRIPTION
**Pull request by richaudience**

### CODE REVIEW

**Comment by richaudience** [comment]
hi guys!
You have any news for the last commit? I need that to be approved as soon as possible.
Cheers!

File: openrtb_ext/imp_richaudience.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package openrtb_ext
 type ExtImpRichaudience struct {
 	Pid         string  `json:"pid"`
 	SupplyType  string  `json:"supplyType"`
+	Keywords    string  `json:"keywords"`
```
Comment (Sonali-More-Xandr) [inline openrtb_ext/imp_richaudience.go:6]: Please add this parameter in bidder-params/richaudience.json

File: adapters/richaudience/richaudience.go [hunk 2]
```diff
@@ -50,7 +48,15 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapte
 			errs = append(errs, &errortypes.BadInput{
 				Message: err.Error(),
 			})
-			return nil, errs
+			continue
+		}
+
+		if request.App != nil {
+			request.App.Keywords = "tagid=" + imp.TagID
```
Comment (gargcreation1992) [inline adapters/richaudience/richaudience.go:55]: Could you please add json test case for above statement?

File: adapters/richaudience/richaudience.go [hunk 3]
```diff
@@ -206,14 +234,12 @@ func parseImpExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpRichaudience, error) {
...
-			if imp.Video != nil {
-				return openrtb_ext.BidTypeVideo
-			}
-			return openrtb_ext.BidTypeBanner
+func getMediaType(impId string, imp openrtb2.Imp) openrtb_ext.BidType {
+	if imp.ID == impId {
+		if imp.Video != nil {
+			return openrtb_ext.BidTypeVideo
 		}
+		return openrtb_ext.BidTypeBanner
```
Comment (gargcreation1992) [inline adapters/richaudience/richaudience.go:242]: This is an anti pattern. Usually, BidType should be set in the response body of the bidder. `response.seatbid[i].bid.ext.prebid.type` is where you should set the BidType in your bidder code and use it to [set TypedBid.](https://github.com/prebid/prebid-server/pull/2669/files#diff-f95090b50cc7f4380eb10531db0822ff98ed8c6b17b99175d224162f1c830267R173)

**Review by gargcreation1992 — COMMENTED** [review]
Suggesting few changes.

File: adapters/richaudience/richaudience.go (hunk 2 above)
Comment (gargcreation1992) [reply adapters/richaudience/richaudience.go:55]: this looks good now.

File: adapters/richaudience/richaudience.go (hunk 3 above)
Comment (gargcreation1992) [reply adapters/richaudience/richaudience.go:242]: Do you plan to resolve this @richaudience ?

**Comment by richaudience** [comment]
For us, it is correct, I don't see the problem, if you explain it to me, we can try to apply some changes to correct it.

**Comment by richaudience** [comment]
I understand what you are saying, at the moment we are not compatible with the multi-format bid offer, so the last change is correct.
If that's all, can you close this task and validate the pull request?
Regards

**Review by gargcreation1992 — APPROVED** [review]

**Review by Sonali-More-Xandr — APPROVED** [review]

**Pull request merged**